students/shawn_degraw/session09/mailroom_oo/cli_main.py

```diff
@@ -1,23 +1,66 @@
 #!/usr/bin/env python
 
-""" Mailroom_oo Main """
+""" Mailroom_oo Main menu, including input and output """
 
-from mailroom_oo.donor_models import *
+from donor_models import *
 
+donor_database = DonorCollection()
 
-def newdonor():
-    """ Adds new donor to the donorDB """
+
+def update_donor_donation():
+    """ Identifies selection as update and not new donor """
+    donor_donation(True)
+
+
+def donor_donation(update_donor=False):
+    """ Adds new donor to the donorDB or updates existing """
 
     while True:
         name = input("Please enter full name or list> ")
         if name == "":
             print("Please enter valid name.\n")
         elif name == "list":
-            print(donor_database.donor_namelist())
+            print('\n'.join(donor_database.collect_data()))
+        elif update_donor and not donor_database.search_name(name):
+            print("Error:  Name not found in donor records.\n")
+            break  # Prevent loop on first entry
         else:
-            donationamount = input("Enter donation amount> ")
-            donor_database.new_donor(name, donationamount)
-            break
+            goforward = input("Is \"{}\" the correct name (Y/N)? ".format(name))
+            if goforward in ["y", "Y"]:
+                while True:
+                    donationamount = input("Enter donation amount> ")
+                    if donor_database.donor_update(name, donationamount):
+                        # print thankyou letter, float will work if donor_upate was True
+                        printthankyou(name, float(donationamount))
+                        break
+                    print("Error:  Please enter a valid donation.")
+                break
+
+
+def printreport():
+    """ Prints report created in the class """
+
+    print(donor_database.create_report())
+    print()
+
+
+def printthankyou(name, donationamount):
```

Comment:
Minor: This is OK, but I think this function could/better be a method of the `DonorCollection` class, or even the `Donor` class.